Newpage

AsianHam opened the pull request:
everything hurts

AsianHam committed:
base new recipe

created a new recipe page with a few form items

created routes to new recipe page

AsianHam committed:
Add New Recipe Initial Completion

Added in remaining forms

Added modal for searching

Submit add recipe information into database

Forms clear on successful POST

Notification prompt showing success or fail

Jacksondr5 commented on package.json:
```diff
@@ -10,6 +10,7 @@
     "antd": "^4.21.3",
     "dotenv": "^16.0.1",
     "dotenv-webpack": "^7.1.1",
+    "mongoose": "^6.4.4",
```
I dont think you need this

Jacksondr5 commented on src/App.css:
```diff
@@ -6,18 +6,26 @@
   margin-bottom: 50px;
-  margin-top: 0px;
+  margin-top: 10px;
+  /* padding-top: 50px; */
```
Go ahead and remove commented code

Jacksondr5 commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,494 @@
+
+  const url = baseUrl + "/recipe";
+
+  const getAllRecipes = async () => {
```
Can you move this into the fetchRecipe.tsx (rename to .ts) file?  Looks like you use the same code in App.tsx

AsianHam committed:
form clear fix

fixed bug where selecting a recipe cleared the form

jackson-miller-tbol commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,495 @@
+  const [loadedRecipes, setLoadedRecipes] = useState<Recipe[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [form] = Form.useForm();
+  const [modalForm] = Form.useForm();
```
If you pass a type to useForm, then you won't have to deal with the forms being `any` types

jackson-miller-tbol commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,495 @@
+
+export default function NewRecipe() {
+  const { TextArea, Search } = Input;
+  const [visible, setVisible] = useState(false);
```
Change this to `modalFormIsVisible`

jackson-miller-tbol commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,495 @@
+  const [visible, setVisible] = useState(false);
+  const [allRecipes, setAllRecipes] = useState<Recipe[]>([]);
+  const [someRecipes, setSomeRecipes] = useState<Recipe[]>([]);
+  const [loadedRecipes, setLoadedRecipes] = useState<Recipe[]>([]);
```
Change this name to `recommendedRecipes`

jackson-miller-tbol commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,495 @@
+  const { TextArea, Search } = Input;
+  const [visible, setVisible] = useState(false);
+  const [allRecipes, setAllRecipes] = useState<Recipe[]>([]);
+  const [someRecipes, setSomeRecipes] = useState<Recipe[]>([]);
```
Rename this to `recommendationSearchResults`

jackson-miller-tbol commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,495 @@
+          {someRecipes.map((recipe, index) => (
+            <div
+              key={index}
+              style={{
```
Can these be moved to CSS?

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+.New {
```
.new {

-css classes shouldn't be capitalized

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+.New {
+  margin-bottom: 50px auto;
```
can I ask what this specifically does? margin-bottom: 50px auto; is the same as margin-bottom: auto; right?

https://www.w3schools.com/cssref/tryit.asp?filename=trycss_margin-bottom2

I tested it here just to double check. I'm pretty sure this isn't a thing. Happy to be wrong though

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+.New {
+  margin-bottom: 50px auto;
+  margin-top: 0px auto;
```
If you have multiple margins/paddings, you can add them in a single line like so: 
p {
  margin: 25px 50px 75px 100px;
}
https://www.w3schools.com/css/css_margin.asp

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+  background-color: #282c34;
+  min-height: 100vh;
+  clear: both;
+  padding: 0;
```
Default padding should be 0, so I don't think this is necessary. Happy to be wrong though.

RhimingMariner replied:
I guess it could inherit something, maybe you're accounting for that?

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+  flex-direction: column;
+}
+
+.New-Body {
```
the css class names leave a bit to be desired, there's lots of different ways to go about naming. I'd look into notation such as .new body {
} if you mean any body classes under new or 
.new > body {
}
If you mean a direct body element under new

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+}
+
+.New-InputColor {
+  background-color: rgb(200, 200, 200);
```
Using hex is the standard - background-color: #9a9a9a;

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+  background-color: rgb(200, 200, 200);
+}
+
+.New-TextColor {
```
if these two classes are the same, you can mark them as the same. CSS is normally the time killer when pages load so if it's not 100% necessary you should condense whatever you can.

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+  border: 0px;
+}
+
+.ant-form-item-control {
```
Same for these 3 on 55, 59, and 63

RhimingMariner commented on src/Cooking.css:
```diff
@@ -64,11 +64,9 @@
 }
 
 .Cooking-StepImage {
-  display: flex;
```
if you delete display: flex, does align-items and justify-content still work? I would assume not, but I'd be happy to be wrong

jackson-miller-tbol commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,495 @@
+      }
+    }
+
+    const current = new Date();
```
Use this library for dates, infinitely easier
https://day.js.org/

RhimingMariner commented on src/NewRecipe.css:
```diff
@@ -0,0 +1,65 @@
+
+.ant-input-group-addon {
+  background-color: darkgray;
+  border: 0px;
```
does the border exist here? Is the 0px value necessary?

AsianHam replied:
![image](https://user-images.githubusercontent.com/25393502/180329064-cc4e9eb8-694e-469b-b2c8-a5297bd8a1bc.png)
 the border indeed exists

![image](https://user-images.githubusercontent.com/25393502/180329085-dab33e1a-90f9-42b1-81a6-856f8ee0684d.png)

jackson-miller-tbol commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,495 @@
+    }/${current.getDate()}/${current.getFullYear()}`;
+
+    const newRecipe: Recipe = {
+      id: allRecipes[allRecipes.length - 1].id + 1,
```
This will cause a bug where if 2 people load the page at the same time and submit recipes (not even at the same time), they will end up with the same ID.  You should always set ID's on the server, where there's less chance of a race condition.

Didn't you say you wanted to switch to mongodb IDs anyway?

AsianHam committed:
pr changes

created interface to handle form output

removed modal

created multi select for recipe recommendations

updated CSS names

moved inline styling to CSS

updated variable names

Jacksondr5 commented on src/CookingPages.tsx:
```diff
@@ -68,6 +69,7 @@ export default function CookingPages() {
               width={300}
               height={200}
               src={cookingRecipe.steps[page].image}
+              style={{ objectFit: "cover" }}
```
Move this to CSS please

Jacksondr5 commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,294 @@
+  }, []);
+
+  const submitRecipe = async (values: AddRecipeForm) => {
+    var image = "";
```
use `let` instead of `var`

Jacksondr5 commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,294 @@
+        directions: step.directions,
+      }));
+
+    const now = dayjs();
+    const date = now.format("MM/DD/YYYY");
```
```suggestion
    const date = dayjs().format("MM/DD/YYYY");
```

Jacksondr5 replied:
You can chain the calls together to make this a one line solution

Jacksondr5 commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,294 @@
+      ingredients: ingredientArray,
+      steps: stepArray,
+    };
+    console.log(newRecipe);
```
Please remove console.logs

Jacksondr5 commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,294 @@
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(newRecipe),
+    };
+    // const response = await PutRecipe(requestOptions);
```
Do you want this to be uncommented?

Jacksondr5 commented on src/NewRecipePage.tsx:
```diff
@@ -0,0 +1,294 @@
+    // }
+  };
+
+  const selectChange = (value: string[]) => {
```
Please rename this to `onRecommendationsChange`

Jacksondr5 commented on src/fetchRecipe.ts:
```diff
@@ -0,0 +1,34 @@
+  throw new Error("The URL environment variable is undefined/missing");
+}
+export async function FetchRecipe<Response>(id?: string) {
+  if (id) {
+    var url = baseUrl + "/recipe/" + id;
+  } else {
+    var url = baseUrl + "/recipe";
+  }
```
```suggestion
  let url = baseUrl + "/recipe";
  if (id) {
    url = url + "/" + id;
  } 
```

Jacksondr5 commented on src/fetchRecipe.ts:
```diff
@@ -0,0 +1,34 @@
+    | { method: string; headers: { "Content-Type": string }; body: string }
+    | undefined
+) {
+  if (!options) {
```
Since this code is called by your code, you don't need a null check here.  Just make `options` a required parameter and TS wont let you call it with a possibly undefined object

AsianHam committed:
pr changes

updated some css

streamlined some logic

removed console logs

uncommented required code
